src/xdocs/checks/annotation/annotationlocation.xml

```diff
@@ -0,0 +1,281 @@
+              <td>6.1</td>
+            </tr>
+            <tr>
+              <td>allowSamelineParameterizedAnnotation</td>
+              <td>Allow one and only parameterized annotation to be located on the same line as
+                  target element.</td>
+              <td><a href="property_types.html#boolean">boolean</a></td>
+              <td><code>false</code></td>
+              <td>6.4</td>
+            </tr>
+            <tr>
+              <td>tokens</td>
+              <td>tokens to check</td>
+              <td>subset of tokens
+                <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CLASS_DEF">
+                    CLASS_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INTERFACE_DEF">
+                    INTERFACE_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#PACKAGE_DEF">
+                    PACKAGE_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_CONSTANT_DEF">
+                    ENUM_CONSTANT_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF">
+                    ENUM_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#METHOD_DEF">
+                    METHOD_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CTOR_DEF">
+                    CTOR_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#VARIABLE_DEF">
+                    VARIABLE_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_DEF">
+                    ANNOTATION_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ANNOTATION_FIELD_DEF">
+                    ANNOTATION_FIELD_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#RECORD_DEF">
+                    RECORD_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#COMPACT_CTOR_DEF">
+                    COMPACT_CTOR_DEF</a>
+                  .
+              </td>
+              <td>
+                <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#CLASS_DEF">
+                    CLASS_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INTERFACE_DEF">
+                    INTERFACE_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#PACKAGE_DEF">
+                    PACKAGE_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_CONSTANT_DEF">
+                    ENUM_CONSTANT_DEF</a>
+                , <a href="apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF">
```

Comment:
This has to be relative to root directory (`src/xdocs`).

Valid URL is https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#ENUM_DEF .
Your now in a sub-directory `checks/annotation`. To keep the same URL, you need to append `../../` to the beginning of this type of URL.

I recommend re-evalulating all changed links in this PR.

Reply:
> I recommend re-evalulating all changed links in this PR.

Already done locally but I am wondering - doing `../../` sounds wrong to me. Using absolute paths (for instance, `/checks/annotation/index.html`) should be easier to maintain.

Reply:
And absolute paths will work on production, but not on S3 and locally it seems.

Reply:
A solution would be to add a `<base>` tag in the headers. I am trying this now:
https://www.w3schools.com/tags/tag_base.asp
> The `<base>` tag specifies the base URL and/or target for all relative URLs in a document.

Reply:
> Using absolute paths (for instance, /checks/annotation/index.html) should be easier to maintain.
> And absolute paths will work on production, but not on S3 and locally it seems

Absolute paths will break documentation versioning since it is in a completely different sub-directory (Example: https://checkstyle.sourceforge.io/version/6.18/ ).

You can only use absolute paths if you need to go somewhere absolutely.

Reply:
Done.